File: tests/test-file-deletion.py
#!/usr/bin/env python3
"""
Test script to verify file deletion behavior
This file can be uploaded and then deleted to test the editor clearing functionality
"""

def main():
    print("Hello! This is a test file for deletion behavior.")
    print("Steps to test:")
    print("1. Upload this file to the application")
    print("2. Open it in the editor")
    print("3. Delete it from the sidebar")
    print("4. Verify the editor content is cleared")
    print("5. Verify the editor shows appropriate message")
    
    # Some sample code to make it interesting
    numbers = [1, 2, 3, 4, 5]
    squared = [x**2 for x in numbers]
    print(f"Original numbers: {numbers}")
    print(f"Squared numbers: {squared}")
    
    return "Test completed successfully!"

if __name__ == "__main__":
    result = main()
    print(result)

File: tests/test-file-deletion.js
// Test file for deletion behavior verification
// This file can be uploaded, opened, and then deleted to test editor clearing

function testDeletionBehavior() {
    console.log("🧪 Testing file deletion behavior");
    console.log("📋 Steps:");
    console.log("1. Upload this file");
    console.log("2. Open it in the editor");
    console.log("3. Delete it from sidebar");
    console.log("4. Check if editor clears properly");
    
    const testData = {
        filename: "test-file-deletion.js",
        purpose: "Test file deletion behavior",
        expected: "Editor should clear when file is deleted"
    };
    
    return testData;
}

// Sample function to make the file more substantial
function calculateSum(numbers) {
    return numbers.reduce((sum, num) => sum + num, 0);
}

// Sample usage
const sampleNumbers = [10, 20, 30, 40, 50];
const total = calculateSum(sampleNumbers);
console.log(`Sum of ${sampleNumbers.join(', ')} = ${total}`);

// Export for testing
module.exports = { testDeletionBehavior, calculateSum };
